core/capabilities/remote/executable/request/client_request.go

```diff
@@ -190,15 +190,39 @@ func (c *ClientRequest) OnMessage(_ context.Context, msg *types.MessageBody) err
 	c.responseReceived[sender] = true
 
 	if msg.Error == types.Error_OK {
-		responseID := sha256.Sum256(msg.Payload)
+		responseID, metadata, err := c.getMessageHashAndMetadata(msg)
```

Comment:
Can you add a comment here explaining why we are having to handle metering differently/separately here?